commands/poll.js

```diff
@@ -0,0 +1,333 @@
+  async execute(interaction) {
+    console.debug("Creating poll...");
+
+    let pollEnded = false;
+    try {
+      const question = interaction.options.getString("question");
+      const pollCloseInSeconds =
+        interaction.options.getInteger("close-in") || 900;
+      const options = [];
+
+      for (let i = 1; i <= MAX_OPTIONS; ++i) {
+        const option = interaction.options.getString(`option-${i}`);
+
+        if (option) {
+          options.push({
+            emoji: `${OPTION_EMOJIS[i - 1]}`,
+            option: option,
+          });
+        }
+      }
+
+      const closePollUnixTimestamp =
+        Math.floor(Date.now() / 1000) + pollCloseInSeconds + DELAY;
+
+      const embed = {
+        color: 0xcf37ca,
+        title: `Poll Question: ${question}`,
+        description: `Created by ${interaction.member.user.tag} - poll expires <t:${closePollUnixTimestamp}:R>`,
+        fields: options.map((info) => {
+          return {
+            name: `${info.emoji}  - ${info.option}`,
+            value: "",
+          };
+        }),
+        footer: {
+          text: "This bot is powered by Xander's money",
+        },
+      };
+
+      const row = new ActionRowBuilder().addComponents(
+        new ButtonBuilder()
+          .setCustomId("close-poll")
+          .setLabel("Close Poll")
+          .setStyle(ButtonStyle.Danger)
+      );
+
+      const pollMessage = await interaction.reply({
+        embeds: [embed],
+        components: [row],
+        fetchReply: true,
+      });
+
+      for (let i = 0; i < options.length; ++i) {
+        await pollMessage.react(`${OPTION_EMOJIS[i]}`);
+      }
+
+      const reactionFilter = (reaction, user) => {
+        return OPTION_EMOJIS.includes(reaction.emoji.name) && !user.bot;
+      };
+
+      const reactionCollector = pollMessage.createReactionCollector({
+        reactionFilter,
+        time: pollCloseInSeconds * 1000,
+        dispose: true,
+      });
+
+      reactionCollector.on("collect", async (reaction, user) => {
+        console.info(`Got reaction ${reaction.emoji.name}  from ${user.tag}`);
+      });
+
+      reactionCollector.on("remove", async (reaction, user) => {
+        console.info(`Removed reaction ${reaction.emoji.name}  by ${user.tag}`);
+      });
+
+      reactionCollector.on("end", (collected) => {
+        console.debug(`Collected ${collected.size} reaction interactions!`);
+      });
+
+      const buttonCollector = pollMessage.createMessageComponentCollector({
+        componentType: ComponentType.Button,
+        time: pollCloseInSeconds * 1000,
+      });
+
+      buttonCollector.on("collect", async (action) => {
+        if (action.user.id === interaction.member.user.id) {
+          const originalEmbed = pollMessage.embeds[0];
+          const computingResultsEmbed = {
+            title: "Computing poll results...",
+            footer: {
+              text: "This bot is powered by Xander's money",
+            },
+          };
```

Comment:
Minor: As both `originalEmbed` and `computingResultsEmbed` are also defined in lines 276 and 278, respectively, we can just define both of these after defining `pollMessage` - preferably after line 169.